wolfSSL-FIPS-Ready/src/section01.md

```diff
@@ -0,0 +1,60 @@
+included in the wolfSSL source tree that one can enable and build. You do not 
+get a FIPS certificate, you are not FIPS approved, but you will be Ready to go 
+through the FIPS process when the time comes. FIPS Ready means that you have 
+included the FIPS code into your build and that you are operating according to 
+the FIPS enforced best practices of default entry point, and conditional 
+algorithm self-tests (CAST) that came with FIPS 140-3. When the time comes, 
+you can get your operating environment tested and added to the wolfSSL Inc 
+FIPS 140-3 certificate for wolfCrypt or submit for a brand new FIPS 140-3 cert 
+from scratch and all the coding work will have been done.
+
+FIPS Ready is open source and dual-licensed. wolfSSL Inc distributes FIPS 
+Ready wolfSSL with the GPLv3 license or can negotiate commercial licensing 
+terms with support if needed beyond the proof of concept phase.
+
+FIPS is a complicated topic. If you have questions after reviewing this 
+document, then just contact us at facts -at- wolfssl  _dot_ calm. The email 
+address is obfuscated for the spiders, but the calm is for the calm you’ll feel 
+when you are FIPS Ready.
+
+## What is Different from a non-FIPS build of wolfSSL? 
+
+The wolfCrypt FIPS API provides wrappers for all the approved algorithm 
+functions that are within the FIPS boundary. The FIPS wrappers can be called 
+directly, or you can keep calling the original APIs; at compile time the API 
+is swapped by the headers so that the FIPS wrappers will be called either way. 
+The FIPS wrapper functions check the status of the internal self-testing before 
+calling the actual function. If the CAST for that specific algorithm has not 
+been run at least one time previously it will run the first time the algorithm 
+is used. Users now have the option to either pre-emptively test algorithms at 
+start up if they wish to avoid the test running at a later time or they can 
+just let the test run when the algorithm is used.
+
+The wolfCrypt FIPS 140-3 ready code has a required power-on self-test (POST) 
+that automatically checks the integrity of the executable in memory, this has 
+not changed since 140-2, only the known answer tests (KATs) for algorithms not 
+used by the POST are now conditional upon use. The executable is organized so 
+the code in the FIPS boundary is adjacent in memory. When the application using 
+the FIPS code starts up, or the shared library is loaded, the default entry 
+point of the library is called, and the POST runs automatically. It has two 
+major parts: the in-core memory check and the known answer tests (KAT) for the 
+algo used by the POST (HMAC-SHA256).
+
+The POST for HMAC-SHA256 is performed first followed by the in-core memory 
+test. The code in memory is hashed with HMAC-SHA256. If the hash matches, the 
+test progresses. Otherwise the FIPS module is placed in an error state and all 
+calls into the module will fail until the integrity hash is updated.
```

Comment:
"all calls into the module will fail until the integrity hash is updated." -> "all calls into the module will fail.  In this case, the FIPS source code must be updated with the correct integrity hash, and only after recompilation can the module successfully initialize."